lib/models/schema.js

```diff
@@ -17,6 +17,12 @@ const MixinSpecificationExtensions = require('../mixins/specification-extensions
  * @returns {Schema}
  */
 class Schema extends Base {
+  constructor (json) {
+    const isBoolSchema = typeof json === 'boolean';
+    super(isBoolSchema ? {} : json);
+    this.isBoolSchema = isBoolSchema;
+  }
```

Comment:
I have the biggest problem here, how to handle it, because we don't support `true/false` schema from the beginning. This solution both adds boolean schema support, and is also backward compatible. Another solution is to return the schema as boolean, but then we must adjust a lot of functions in another models, which returns array of schemas or single schema. Any thoughts?

Reply:
I think it makes sense to keep **all** schemas as `Schema` object types. Also, if you return pure boolean types here, checks would need always to be performed as a strict comparison (`===` / `!===`), which sounds pretty error-prone. `schema.isBoolSchema` sounds like a more readable version and also less error-prone since we can make a statement on "all schemas are represented as `Schema` objects.

Reply:
Yes, you're right :)

Reply:
Yep true, I can see it as benefit as well 👍